packages/dosespot-react/src/DoseSpotNewMedicationForm.tsx

```diff
@@ -0,0 +1,154 @@
+import { Box, Button, Divider, Group, NumberInput, Stack, Text, TextInput } from '@mantine/core';
+import { NDC, RXNORM } from '@medplum/core';
+import { Bundle, MedicationKnowledge } from '@medplum/fhirtypes';
+import React, { useState } from 'react';
+import { DOSESPOT_DISPENSABLE_DRUG_ID_SYSTEM, DOSESPOT_REFILLS_SYSTEM } from './common';
+import { DoseSpotMedicationSelect } from './DoseSpotMedicationSelect';
+
+export interface DoseSpotNewMedicationFormProps {
+  readonly searchMedications: (term: string) => Promise<Bundle<MedicationKnowledge> | undefined>;
+  readonly addFavoriteMedication: (medication: MedicationKnowledge) => Promise<void>;
+  readonly loading?: boolean;
+}
+
+interface FormData {
+  quantity: number;
+  refills: number;
+  directions: string;
+}
+
+export function DoseSpotNewMedicationForm(props: DoseSpotNewMedicationFormProps): React.JSX.Element {
+  const { searchMedications, addFavoriteMedication, loading = false } = props;
+  const [formData, setFormData] = useState<FormData>({
+    quantity: 1,
+    refills: 0,
+    directions: '',
+  });
+  const [medication, setMedication] = useState<MedicationKnowledge | null>(null);
+
+  const handleAddFavorite = async (): Promise<void> => {
+    // Create the updated medication with proper FHIR structure
+    const completedMedication: MedicationKnowledge = {
+      ...medication,
+      resourceType: 'MedicationKnowledge',
+      // Use amount field for quantity
+      amount: {
+        value: formData.quantity,
+        unit: 'units',
+      },
+      extension: [
+        {
+          url: DOSESPOT_REFILLS_SYSTEM,
+          valueString: formData.refills.toString(),
+        },
+      ],
```

Comment:
I don't want to introduce an extensions for now, as we talked about updating this to use `ActivityDefinition` in the future